lumicks/pylake/detail/image.py

```diff
@@ -217,8 +222,8 @@ def reconstruct_image(data, infowave, shape, reduce=np.sum):
     #   pixel_sizes = np.diff(np.flatnonzero(infowave == InfowaveCode.pixel_boundary))
     # But for now we assume that every pixel consists of the same number of samples
     pixel_size = np.argmax(subset) + 1
-    resized_data = np.zeros(round_up(subset.size, pixel_size))
-    resized_data[: subset.size] = data[valid_idx]
+    resized_data = np.zeros(round_down(subset.size, pixel_size), dtype=data.dtype)
+    resized_data = data[valid_idx][: resized_data.size]
```

Comment:
You don't need the `np.zeros()` line at all. All the zeros will be overwritten by values from `data`. You could just do this:
```py
resized_data = data[valid_idx][:round_down(subset.size, pixel_size)]
```

Reply:
Right. 👍